README.md

```diff
@@ -1,168 +1,156 @@
+or the HTML `<meta>` element
 
-This strongly indicates an opt-in in the HTML document, such as a `<meta>` tag, attribute of the `<html>` element, or other affordance entirely within the HTML response body.
+```html
+<meta http-equiv="Supports-Loading-Mode"
+      content="uncredentialed-prerender">
+```
 
-### Document policy
+As in the last example, based on its heuristics or informed by prerender triggers, the browser can attempt to prerender these linked-to news articles. Since they are cross-origin, however, the process is more restricted. The initial fetch, as well as any subresource fetches, are performed without credentials. If the response that comes back does not have the opt-in, then the result is discarded.
 
-[Document policy][document-policy] provides a related mechanism, which allows a document to provide a set of behavior changes and restrictions it wishes to apply to itself in an HTTP response header, `Document-Policy`. The request may contain a `Sec-Required-Document-Policy` header, which indicates the minimum level of strictness for each policy that is required -- otherwise the load will fail.
+If the opt-in is present, then the resulting document is loaded into a cross-origin prerendering browsing context. Content there is restricted more heavily; not only are disruptive APIs prevented from working, but also storage access is initially not available. If the news site intends to personalize itself, e.g. to reflect subscriber status, then it would use code such as the following:
 
-Document policy does not support a mechanism for setting policy inside the response body, which we believe to be important to ease of deployment by authors. It might be possible to extend document policy to also support this, but there would be significant added complexity in document policy, such as algorithms for combining policy declarations in both places and deferring loads in contexts with a required document policy until it can be determined that the merged policy is suitable. These problems are somewhat similar in the special case this proposal discusses than in the general case of document policy.
+```js
+function afterPrerendering() {
+  // grab user data from cookies/IndexedDB
+  // update the UI
+}
 
-`Sec-Required-Document-Policy` would not replace the need for a [request header](#request-header), since this is settable by content that is not genuinely prerendering, e.g. via `<iframe policy>`.
+if (!document.loadingMode || document.loadingMode.type === 'default') {
+  afterPrerendering();
+} else {
+  document.addEventListener('loadingmodeechange', () => {
+    if (document.loadingMode.type === 'default') {
+      afterPrerendering();
+    }
+  });
+}
```

Comment:
Noting, of course, that I would prefer to make something like privacycg/storage-access#55 a default choice for cases like this that really just care about whether they can see their storage yet. Pending ongoing discussions about how storage access transitions will work in other contexts, also on that repo.

The detailed loadingmodechange API can hopefully be used mainly by stuff that actually cares about why they can't touch storage and other restrictions that apply in a prerendered browsing context. Very bikesheddable.